broker/duplicate_certs.py

```diff
@@ -57,16 +57,35 @@ def delete_iam_server_certificate(certificate_name):
     except iam_govcloud.exceptions.NoSuchEntityException as e:
         logger.info(f"IAM certificate {certificate_name} does not exist: {e}, continuing")
 
+def get_listener_cert_arns(listener_arn, alb=alb):
+    response = alb.describe_listener_certificates(
+        ListenerArn=listener_arn,
+    )
+    listener_cert_arns = [cert["CertificateArn"] for cert in response["Certificates"]]
+    return listener_cert_arns
+
+def remove_certificate_from_listener_and_verify_removal(listener_arn, certificate_arn, alb=alb):
+    alb.remove_listener_certificates(
+        ListenerArn=listener_arn,
+        Certificates=[{"CertificateArn": certificate_arn}]
+    )
+    attempts = 0
+    while True:
+        if (attempts == 10):
+            logger.info(f"Could not verify certificate {certificate_arn} was removed from listener {listener_arn} after 10 tries, giving up")
+            break
+        listener_cert_arns = get_listener_cert_arns(listener_arn, alb=alb)
+        if certificate_arn not in listener_cert_arns:
+            break
+        attempts += 1
+    logger.info(f"Removed certificate {certificate_arn} from listener {listener_arn}")
```

Comment:
If I may suggest a simplification:

```suggestion
    for _ in range(10):
        listener_cert_arns = get_listener_cert_arns(listener_arn, alb=alb)
        if certificate_arn not in listener_cert_arns:
            logger.info(f"Removed certificate {certificate_arn} from listener {listener_arn}")
            return
    logger.info(f"Could not verify certificate {certificate_arn} was removed from listener {listener_arn} after 10 tries, giving up")
```

Also, did you consider using a different log level for the failure case? Not sure if it provides any value, but thought I'd ask.